app/services/fastboot.js

```diff
@@ -46,5 +46,8 @@ export default Ember.Service.extend({
 
   isFastBoot: computed(function() {
     return typeof FastBoot !== 'undefined';
-  })
+  }),
+  deferRendering(promise) {
```

Comment:
don't we more typically have hooks, the when invoked do.

new Promise(resolve => resolve(cb());

Passing promises around like this doesn't really align with other hooks, like model routes hooks, or adapter hooks.